client/beefy/src/worker.rs

```diff
@@ -500,19 +500,27 @@ where
 
 			self.on_demand_justifications.cancel_requests_older_than(block_num);
```

Comment:
Something for @acatangiu. Why do we use the block number here as part of the commitment and not the block hash?

Reply:
All BEEFY voting and justifications/proofs happen on GRANDPA-finalized blocks, therefore block numbers also uniquely identify blocks. (My best guess) block numbers were used instead of hashes to lower cognitive complexity when following BEEFY proofs/progress.

Reply:
> therefore block numbers are enough to uniquely identify blocks.

Yeah I know.



> (My best guess) block numbers were initially used instead of hashes to lower cognitive complexity when following BEEFY proofs/progress

Hmm okay :P